src/control/client/network.go

```diff
@@ -0,0 +1,173 @@
+//
+// (C) Copyright 2018-2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+
+	ctlpb "github.com/daos-stack/daos/src/control/common/proto/ctl"
+)
+
+const (
+	networkScanTimeout = 10 * time.Second
+)
+
+type ProviderListResult struct {
+	providerList string
+}
+
+type NetworkScanResult struct {
+	provider string
+	device   string
+	numanode uint
+	err      error
+}
+
+type NetworkScanResultMap map[string][]NetworkScanResult
+
+func (nsrm NetworkScanResultMap) String() string {
+	var buf bytes.Buffer
+	servers := make([]string, 0, len(nsrm))
+
+	for server := range nsrm {
+		servers = append(servers, server)
+	}
+	sort.Strings(servers)
+
+	for _, server := range servers {
+		fmt.Fprintf(&buf, "%s:\n%s\n", server, nsrm[server])
+	}
+
+	return buf.String()
+}
+
+func (nsr NetworkScanResult) String() string {
+	var buf bytes.Buffer
+	if nsr.err != nil {
+		fmt.Fprintf(&buf, "\n\tError: %v\n", nsr.err)
+		return buf.String()
+	}
+	fmt.Fprintf(&buf, "\n\tfabric_iface: %s\n\tprovider: %s\n\tpinned_numa_node: %d\n", nsr.device, nsr.provider, nsr.numanode)
+	return buf.String()
+}
+
+func (c *connList) NetworkListProviders() ResultMap {
+	c.log.Debugf("NetworkListProviders() Received")
+	return c.makeRequests(&ctlpb.ProviderListRequest{}, providerListRequest)
+}
+
+func providerListRequest(mc Control, req interface{}, ch chan ClientResult) {
+	sRes := ProviderListResult{}
```

Comment:
It seems like you're only ever using sRes.providerList, so why not just dispense with the struct? Is it an artifact of an older approach or was there a thought that you would want to use the struct in the future?

Reply:
I patterned it after other results structs and kept it that way to preserve the style of the others.  There's no technical reason why it needs to be a struct given it has only one string variable.  Do you want it changed?

Reply:
| Do you want it changed?

It's your code... It's not technically wrong, just kind of odd. For the record, the request/result struct pattern doesn't really apply when you're not crossing API/package boundaries, and definitely not within the scope of a single function. The goal is just to encourage self-documenting and more maintainable code.